openlibrary/coverstore/archive.py

```diff
@@ -89,6 +91,50 @@ def close(self):
 idx = id
 
 
+def is_uploaded(item, f):
+    command = fr'ia list {item} | grep "{f}\.[tar|index]" | wc -l'
+    result = subprocess.run(command, shell=True, capture_output=True)
+    output = result.stdout.decode().strip()
+    return int(output) == 2
+
+
+def audit(group_id, chunk_ids=(0, 100), sizes=('', 's', 'm', 'l'), upload=False):
+    """Check which cover batches have been uploaded to archive.org.
+
+    Checks the archive.org items pertaining to this `group` of up to
+    1 million images (4-digit e.g. 0008) for each specified size and verify
+    that all the chunks (within specified range) and their .indices + .tars (of 10k images, 2-digit
+    e.g. 81) have been successfully uploaded.
+
+    {size}_covers_{group}_{chunk}:
+    group_id - 4 digit, batches of 1M, 0000 to 9999M
+    chunk_ids - (min, max) chunk_id range or max_chunk_id; 2 digit, batch of 10k from [00, 99]
+
+    """
+    scope = chunk_ids if isinstance(chunk_ids, range) else range(0, chunk_ids)
+    for size in sizes:
+        prefix = f"{size}_" if size else ''
+        item = f"{prefix}covers_{primary}"
+        files = [f"{prefix}covers_0008_{'%02i' % i}" for i in scope]
```

Comment:
Keep this is a generator, since we don't need this as a literal list.
```suggestion
        files = (f"{prefix}covers_0008_{i:02}" for i in scope)
```

Reply:
resolved in separate commit